MIPs/mip-mc-handshake.md

```diff
@@ -0,0 +1,255 @@
+## Implementation
+Implementing this proposal involves a significant modification to MetaMask's JSON-RPC request-handling code to support the new `provider_authorize` method, adjustments to the permissioning system, and updates to the UI components for user consent and management of dapp permissions.
+
+## Developer Adoption Considerations
+
+[Explain any considerations that developers should take into account when adopting this proposal. For example, how will it affect compatibility, and what changes may need to be made to accommodate the proposal?]
+
+Backward compatibility will be maintained through the existing Ethereum Provider API. However, the Multichain API WILL NOT be available through the existing Ethereum Provider API. Because many developers rely on third-party libraries to connect their dapps with wallets, a some mapping logic that allows them to keep their "single chain" code as-is while actually hitting the multichain API may facilitate more rapid adoption. 
+
+Developers should expect most new improvements to the Wallet API to be delivered only though the multichain API, NOT the Ethereum Provider API.
+
+## User Experience Considerations
+
+Calling the `provider_authorize` method with `optionalScopes` that include any eip155 prefixed scopes will trigger the following sequence:
+
+```mermaid
+sequenceDiagram
+    app->>+MetaMask: provider_authorize Request
+    actor User
+    MetaMask->>User: Network Permissions
+    activate User
+    MetaMask->>User: Acccount Permissions
+    User-->>MetaMask: Confirm
+    deactivate User
+    MetaMask-->>-app: provider_authorize Response
+```
+
+## Privacy Considerations
+This proposal raises important privacy considerations, including the need to avoid data leaking and the challenge of obtaining genuine user consent. It underscores the importance of preserving user anonymity and the sensitivites involved in determining authorizations. Identifying and mitigating these issues is crucial for protecting user privacy during multichain interactions, prompting a careful evaluation of how best to balance functionality with privacy concerns.
+
+## Security Considerations
+This proposal is an opportunity to further incorporate the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) in MetaMask.
+
+The proposal also brings to light some security considerations critical to multichain interactions. These include the challenges of ensuring robust authentication and authorization and the importance that users understand the chain with which they are interacting. Identifying and addressing these issues is vital for safeguarding users against the evolving landscape of security threats in the blockchain ecosystem.
+
+## Committed Developers
+MetaMask Portfolio
+MetaMask Bridge
+
+## Open Issues
+List of significant open issues that require resolution in order for this MIP to be ready to be moved to the `Review` stage
+- Add OpenRPC specs for the provider_authorize method
```

Comment:
feel free to tag me in this PR if i can be of any service, or just to provide implementer feedback on the CAIP-25 spec qua spec!

Reply:
Here is my initial attempt at speccing out `provider_authorize` we can start from:

https://github.com/ChainAgnostic/CAIPs/pull/212/files